pandas/io/parsers.py

```diff
@@ -847,12 +847,18 @@ def _validate_usecols_arg(usecols):
     or strings (column by name). Raises a ValueError
     if that is not the case.
     """
+    msg = ("The elements of 'usecols' must "
+           "either be all strings or all integers")
+
     if usecols is not None:
         usecols_dtype = lib.infer_dtype(usecols)
-        if usecols_dtype not in ('integer', 'string'):
-            raise ValueError(("The elements of 'usecols' "
-                              "must either be all strings "
-                              "or all integers"))
+
+        if usecols_dtype == 'mixed':
```

Comment:
@jreback : that's in conflict with what @sinhrks is suggesting <a href="https://github.com/pydata/pandas/pull/13233/files/71f5139f14b38a1a3c1d9fd7d9219947399faec0#r63986201">below</a>.  Perhaps that should be sorted out?